Ecal/src/Ecal/EcalVetoProcessor.cxx

```diff
@@ -1042,13 +1044,21 @@ void EcalVetoProcessor::produce(framework::Event &event) {
           }
         }
 
-        // Perform "linreg" on selected points:
-        TDecompSVD svdObj = TDecompSVD(hdt);
+        // Perform "linreg" on selected points
+        // Calculate the determinant of the matrix
+        double determinant =
+            hdt(0, 0) * (hdt(1, 1) * hdt(2, 2) - hdt(1, 2) * hdt(2, 1)) -
+            hdt(0, 1) * (hdt(1, 0) * hdt(2, 2) - hdt(1, 2) * hdt(2, 0)) +
+            hdt(0, 2) * (hdt(1, 0) * hdt(2, 1) - hdt(1, 1) * hdt(2, 0));
+        // Exit early if the matrix is singular (i.e. det = 0)
+        if (determinant == 0) continue;
```

Comment:
Any concerns on numerical precision? (Should we check `abs(determinant) < 1e-6` or something instead?)

Reply:
I was thinking about this too, but looking at some specific cases, it's always zero. And since in the matrix we have this subtraction of the mean, it can lead to small numbers, which then means the determinant can be a small number. And then in a few lines down, we are calling ROOT's decompose that will kill the events like that, so I'd prefer to have this as is, just to kill the diabolical events.

Reply:
(i.e. mathematically it's possible to divide with  1e-6, while it's not with 0)